File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
            substring = ""
            maxLen = 0
            for i in s:
                if i not in substring:
                    substring = substring + i
                    maxLen = max(maxLen, len(substring))
                else:
                    substring = substring.split(i)[1] + i
            return maxLen

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.go
func lengthOfLongestSubstring(s string) int {
    ans := 0
    m := map[byte]bool{}
    l, r := 0, 0
    for r < len(s) {
        if _, ok := m[s[r]]; ok {
            delete(m, s[l])
            l += 1
        } else {
            m[s[r]] = true
            r += 1
            ans = max(ans, r - l)
        }
    }

    return ans
}